testing.py

```diff
@@ -300,7 +297,240 @@ def printOut(text):
+            prodFlowStatisticsFileNotFound = True
+        
+        if (testFlowStatisticsFileNotFound is False and prodFlowStatisticsFileNotFound is False):
+            # Compare the files
+            testStatisticGroups = json.load(testFlowStatisticsFile)
+            testDictionary = {}
+            for testStatisticGroup in testStatisticGroups:
+                testDictionary[testStatisticGroup['StatisticGroupID']] = {}
+                for testRegressionRegion in testStatisticGroup['RegressionRegions']:
+                    testDictionary[testStatisticGroup['StatisticGroupID']][testRegressionRegion['Code']] = {}
+                    for testResult in testRegressionRegion["Results"]:
+                        testDictionary[testStatisticGroup['StatisticGroupID']][testRegressionRegion['Code']][testResult["code"]] = testResult["Value"]
+
+            prodStatisticGroups = json.load(prodFlowStatisticsFile)
+            for prodStatisticGroup in prodStatisticGroups:
+                for prodRegressionRegion in prodStatisticGroup['RegressionRegions']:
+                    for prodResult in prodRegressionRegion["Results"]:
+                        if testDictionary[prodStatisticGroup['StatisticGroupID']][prodRegressionRegion['Code']][prodResult["code"]] != prodResult["Value"]:
+                            numberFlowStatisticsNotEqual += 1
+                            dataRow = [folderName, "TEST", productionServer, region, siteID, "FlowStatistics", prodResult["code"], str(testDictionary[prodStatisticGroup['StatisticGroupID']][prodRegressionRegion['Code']][prodResult["code"]]), str(prodResult["Value"])]
+                            comparisonTestProdDifferencesFileWriter.writerow(dataRow)
+                            comparisonTestProdDifferencesFile.flush()
+                            comparisonSummaryFileWriter.writerow(dataRow)
+                            comparisonSummaryFile.flush()
+                        dataRow = [folderName, "TEST", productionServer, region, siteID, "FlowStatistics", prodResult["code"], str(testDictionary[prodStatisticGroup['StatisticGroupID']][prodRegressionRegion['Code']][prodResult["code"]]), str(prodResult["Value"]), str(testDictionary[prodStatisticGroup['StatisticGroupID']][prodRegressionRegion['Code']][prodResult["code"]] == prodResult["Value"])]
+                        comparisonTestProdFileWriter.writerow(dataRow)
+                        comparisonTestProdFile.flush()
+            if numberFlowStatisticsNotEqual > 0:
+                printOut("Number of Flow Statistics not equal: " + str(numberFlowStatisticsNotEqual))
+            else:
+                printOut("All Flow Statistics are equal.")
+        elif (testFlowStatisticsFileNotFound is True and prodFlowStatisticsFileNotFound is True):
+            printOut("Flow Statistics are equal.")
+            dataRow = [folderName, "TEST", productionServer, region, siteID, "FlowStatistics", "", "File not found", "File not found", str(True)]
```

Comment:
extra comma here

Reply:
Testing-2022-05-12-12-44-47,TEST,PRODWEBB,WI,05401800,FlowStatistics**,,**File not found,File not found,True

Reply:
The comma is there because there is no "Description" for FlowStatistics in the case that there isn't a Flow Statistics file for both Test and the Prod machine. The "Description" column is usually used to describe the specific flow statistic that is being compared. Definitely a bit confusing! But it's working as expected.